pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,143 @@ $title: Use custom JavaScript in AMP pages
 
-The DOM inside `<amp-script>` tags should be almost identical before and after initialization.
+A handful of synchronous DOM API methods are replaced with alternatives that return a Promise. For example,  `getBoundingClientRect()` is replaced by `getBoundingClientRectAsync()`.
 
-For example, if starting with the code below:
-```html
-<text> Hello world </text>
-```
-Worker DOM permits minor changes in structure but not content:
+To maintain AMP's performance guarantees, certain actions [are restricted](../../../documentation/components/reference/amp-script.md#user-gestures). Your code is discouraged from making potentially distracting mutations unless they follow a user interaction. You can't add stylesheets or additional scripts to the DOM, and [`importScripts()`](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts) is not supported.
 
-```html
- <text>Hello </text><text>world</text>
-```
 
-## DOM manipulation
+## Preact and other frameworks
+
+Given this partial DOM API support, how is it best to approach `amp-script`? Here are two possibilities:
+
+**1) Know what's supported.** Get to know [the rich set of supported APIs](https://github.com/ampproject/worker-dom/blob/main/web_compat_table.md). You need to think about JavaScript differently - more as a concise set of tools.
+
+**2) Use Preact.** Since the team couldn't rebuild all the DOM APIs, they focused on the limited set of APIs used by [Preact](https://preactjs.com/) and [React](https://reactjs.org/). If you use one of these frameworks, you should be able to build elaborate interactions with much less worry about what's supported. That said, React bundles often exceed `amp-script`'s [150K limit](../../../documentation/components/reference/amp-script.md#size-of-javascript-code). For this reason, Preact is strongly recommended.
+
+`amp-script` has been tested with frameworks like [Vue](https://vuejs.org/), [Angular](https://angularjs.org/), [Aurelia](https://aurelia.io/), and [lit-html](https://lit-html.polymer-project.org/), but less extensively. If you find a gap, please [file an issue](https://github.com/ampproject/worker-dom/issues) - or, better still, [submit a pull request](https://github.com/ampproject/worker-dom/pulls).
+
+Since `amp-script` cannot exhaustively support the DOM API, simply copying a library like [jQuery](https://jquery.com/) into an `<amp-script>` component will not work.
+
+## How should I use amp-script?
+
+`amp-script` was created to help you fill the gaps in your AMP webpage's functionality. One can't simply copy existing JavaScript into `amp-script` and expect it to work without modification. Instead, `amp-script` takes care of logic and interactions that existing AMP components don't provide.
 
-For user experience and security reasons, `amp-script` enforces DOM manipulation restrictions.
+The following are a few excellent use cases for `amp-script`.
```

Comment:
I think it's important to let people know that they can't just copy existing JS into amp-script, since they're always trying this. I know we want to be positive, but I also want us to save developers time. Tried to rephrase all this so it's more informative for someone who's just skimming the doc - and so it's more guide-like.